examples/step-87/doc/intro.dox

```diff
@@ -0,0 +1,268 @@
+The subsequent fine search by each process determines whether the processes
+actually own these points by the sequence of request
+("Does the process own the point?") and  answer ("Yes."/"No.").
+Processes might post any number of point requests and communicate with
+any process. We propose to collect the point requests to a process to
+use the dynamic, sparse, scalable consensus-based communication
+algorithms @cite hoefler2010scalable, and to consider the obtained information to
+set up point-to-point communication patterns.
+
+<h3>Implementation: Utilities::MPI::RemotePointEvaluation</h3>
+
+The algorithm described above is implemented in Utilities::MPI::RemotePointEvaluation
+(short: "RPE") and related classes/functions. In this section,
+basic functionalities are briefly summarized. Their advanced
+capabilities will be shown subsequently based on concrete application cases.
+
+The following code snippet shows the setup steps for the communication pattern:
+@code
+std::vector<Point<dim>> points; // ... (filling of points not shown)
+
+RemotePointEvaluation<dim> rpe;
+rpe.reinit(points, triangulation, mapping);
+@endcode
+
+All what is needed is a list of evaluation points and the mesh with a mapping.
+
+The following code snippet shows the evaluation steps:
+@code
+const std::function<void(const ArrayView<T> &, const CellData &)>
+  evaluation_function;
+
+std::vector<T> output;
+rpe.evaluate_and_process(output, evaluation_function);
+
+@endcode
+
+The user provides a function that processes the locally owned points.
+These values are communicated by Utilities::MPI::RemotePointEvaluation.
+
+The relevant class during the local evaluation is
+Utilities::MPI::RemotePointEvaluation::CellData. It allows to loop over
+cells that surround the points. On these cells,
+a cell iterator and the positions in the reference cell of the
+request points can be queried. Furthermore, this class provides
+controlled access to the output vector of the
+Utilities::MPI::RemotePointEvaluation::evaluate_and_process() function.
+@code
+for (const auto cell_index : cell_data.cell_indices())
+  {
+    const auto cell         = cell_data.get_active_cell_iterator(cell_index);
+    const auto unit_points  = cell_data.get_unit_points(cell_index);
+    const auto local_output = cell_data.get_data_view(cell_index, output);
+  }
+@endcode
+
+The functions
+@code
+const auto evaluated_values =
+  VectorTools::point_values<n_components>(rpe, dof_handler, vector);
+
+const auto evaluated_gradients =
+  VectorTools::point_gradients<n_components>(rpe, dof_handler, vector);
+@endcode
+
+evaluate the values and gradients of a solution defined by DoFHandler and a
+vector at the request points. Internally, a lambda function is passed to
+Utilities::MPI::RemotePointEvaluation.
+Additionally it handles the special case, if points belong to multiple cells
+by taking, e.g., the average, the minimum, or
+the maximum via an optional argument of type EvaluationFlags::EvaluationFlags.
```

Comment:
How could a point belong to multiple cells? It would be interesting to add a precision here on when this could occur (e.g. mortar methods come to mind?)

Reply:
In the case that a point lies on a vertex/edge/face, the point can be assigned to multiple cells (depending on some tolerances). This can be turned of by a parameter (which is also used when inserting particles). This is a useful feature when working with DG or you need to test with the shape functions (which are defined over multiple cells), as in the case of mortar methods or sharp-interface methods.

Reply:
Yeah that's what I thought. I would explain here at least when that can occur or why it would occur. This way the reader is aware that sometimes this is a desirable properties (e.g. Mortar methods) or sometimes this is more of an artifact.

Reply:
I've added a sentence.